packages/manager/apps/web/client/app/hosting/email/hosting-email.service.js

```diff
@@ -0,0 +1,93 @@
+import filter from 'lodash/filter';
+import isEmpty from 'lodash/isEmpty';
+
+import {
+  WEB_HOSTING_SERVICE_NAME,
+  WEB_HOSTING_EMAIL_FAMILY_NAME,
+  WEB_HOSTING_EMAIL_PRICING_MODE_DEFAULT,
+  WEB_HOSTING_CONF_ITEM_LEGACY_DOMAIN,
+} from './activate/activate.constants';
+
+export default class HostingEmailService {
+  /* @ngInject */
+  constructor($translate, OrderService, OvhApiHostingWeb, OvhApiOrder) {
+    this.$translate = $translate;
+    this.OrderService = OrderService;
+    this.OvhApiHostingWebEmailOption = OvhApiHostingWeb.EmailOption().v6();
+    this.OvhApiOrderCart = OvhApiOrder.Cart().v6();
+  }
+
+  getEmailOfferDetails(serviceName) {
+    return this.OvhApiHostingWebEmailOption.query({ serviceName }).$promise;
+  }
+
+  createCart(ovhSubsidiary) {
+    return this.OrderService.createNewCart(ovhSubsidiary);
```

Comment:
Maybe you could call directly `OrderService.createNewCart`, I'm not too sure about the benefits of having an abstraction in `HostingEmailService` since `OrderService` already provides it

Reply:
I need abstraction in all other cart methods apart from this one. To be consistency, I added it as well.